File: Java/listas.java
import java.util.*;

public class listas {
    public static void main(String[] args) {
        List<Integer> mi_lista = Arrays.asList(34, 12, 465, 67, 23, 89, 5, 90, 56, 78);
        
        // Ordenamos la lista (creando una copia mutable)
        List<Integer> lista_ordenada = new ArrayList<>(mi_lista);
        Collections.sort(lista_ordenada);
        System.out.println("Lista ordenada: " + lista_ordenada);
        
        // Extraemos un rango (índices 2 a 6)
        List<Integer> sub_lista = lista_ordenada.subList(2, 7);
        System.out.println("Sublista (índices 2 al 6): " + sub_lista);
        
        // Nueva lista con elementos elevados al cuadrado
        List<Integer> nueva_lista = new ArrayList<>();
        for (int num : sub_lista) {
            nueva_lista.add(num * num);
        }
        System.out.println("Nueva lista con elementos elevados al cuadrado: " + nueva_lista);
        
        // Lista modificada: reemplazar el tercer elemento por 100
        List<Integer> lista_modificada = new ArrayList<>(mi_lista);
        if (lista_modificada.size() >= 3)
            lista_modificada.set(2, 100);
        System.out.println("Lista modificada: " + lista_modificada);
    }
}

File: C++/listas.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    vector<int> mi_lista = {34, 12, 465, 67, 23, 89, 5, 90, 56, 78};
    
    // Ordenamos la lista
    vector<int> lista_ordenada = mi_lista;
    sort(lista_ordenada.begin(), lista_ordenada.end());
    cout << "Lista ordenada: ";
    for (int n : lista_ordenada)
        cout << n << " ";
    cout << endl;
    
    // Extraemos un rango de la lista (índices 2 a 6)
    vector<int> sub_lista(lista_ordenada.begin() + 2, lista_ordenada.begin() + 7);
    cout << "Sublista (indices 2 al 6): ";
    for (int n : sub_lista)
        cout << n << " ";
    cout << endl;
    
    // Nueva lista con elementos elevados al cuadrado
    vector<int> nueva_lista;
    for (int n : sub_lista)
        nueva_lista.push_back(n * n);
    cout << "Nueva lista con elementos elevados al cuadrado: ";
    for (int n : nueva_lista)
        cout << n << " ";
    cout << endl;
    
    // Lista modificada: reemplazamos el tercer elemento por 100
    vector<int> lista_modificada = mi_lista;
    if (lista_modificada.size() >= 3)
        lista_modificada[2] = 100;
    cout << "Lista modificada: ";
    for (int n : lista_modificada)
        cout << n << " ";
    cout << endl;
    
    return 0;
}
